src/test/java/net/snowflake/client/diagnostic/DiagnosticContextLatestIT.java

```diff
@@ -0,0 +1,275 @@
+        diagnosticContext.getHttpProxyHost().isEmpty());
+    assertEquals(
+        "getHttpProxyPort() must return -1 in the absence of proxy configuration",
+        -1,
+        diagnosticContext.getHttpProxyPort());
+    assertTrue(
+        "getHttpsProxyHost() must return an empty string in the absence of proxy configuration",
+        diagnosticContext.getHttpsProxyHost().isEmpty());
+    assertEquals(
+        "getHttpsProxyPort() must return -1 in the absence of proxy configuration",
+        -1,
+        diagnosticContext.getHttpsProxyPort());
+    assertTrue(
+        "getHttpNonProxyHosts() must return an empty string in the absence of proxy configuration",
+        diagnosticContext.getHttpNonProxyHosts().isEmpty());
+  }
+
+  @Test
+  public void testProxyConfigSetOnJvm() {
+    System.setProperty(HTTP_PROXY_HOST, "http.proxyHost.com");
+    System.setProperty(HTTP_PROXY_PORT, "8080");
+    System.setProperty(HTTPS_PROXY_HOST, "https.proxyHost.com");
+    System.setProperty(HTTPS_PROXY_PORT, "8083");
+    System.setProperty(HTTP_NON_PROXY_HOSTS, "*.domain.com|localhost");
+
+    Map<SFSessionProperty, Object> connectionPropertiesMap = new HashMap<>();
+
+    DiagnosticContext diagnosticContext = new DiagnosticContext(connectionPropertiesMap);
+
+    System.clearProperty(HTTP_PROXY_HOST);
+    System.clearProperty(HTTP_PROXY_PORT);
+    System.clearProperty(HTTPS_PROXY_HOST);
+    System.clearProperty(HTTPS_PROXY_PORT);
+    System.clearProperty(HTTP_NON_PROXY_HOSTS);
+
+    assertTrue(diagnosticContext.isProxyEnabled());
+    assertTrue(diagnosticContext.isProxyEnabledOnJvm());
+    assertEquals(diagnosticContext.getHttpProxyHost(), "http.proxyHost.com");
+    assertEquals(diagnosticContext.getHttpProxyPort(), 8080);
+    assertEquals(diagnosticContext.getHttpsProxyHost(), "https.proxyHost.com");
+    assertEquals(diagnosticContext.getHttpsProxyPort(), 8083);
+    assertEquals(diagnosticContext.getHttpNonProxyHosts(), "*.domain.com|localhost");
+  }
+
+  /*
+  If Proxy settings are passed using JVM arguments and connection parameters
+  then the connection parameters take precedence.
+   */
+  @Test
+  public void testProxyOverrideWithConnectionParameter() {
+
+    System.setProperty(HTTP_PROXY_HOST, "http.proxyHost.com");
+    System.setProperty(HTTP_PROXY_PORT, "8080");
+    System.setProperty(HTTPS_PROXY_HOST, "https.proxyHost.com");
+    System.setProperty(HTTPS_PROXY_PORT, "8083");
+    System.setProperty(HTTP_NON_PROXY_HOSTS, "*.domain.com|localhost");
+
+    Map<SFSessionProperty, Object> connectionPropertiesMap = new HashMap<>();
+
+    connectionPropertiesMap.put(SFSessionProperty.PROXY_HOST, "override.proxyHost.com");
+    connectionPropertiesMap.put(SFSessionProperty.PROXY_PORT, "80");
+    connectionPropertiesMap.put(SFSessionProperty.NON_PROXY_HOSTS, "*.new_domain.com|localhost");
+
+    DiagnosticContext diagnosticContext = new DiagnosticContext(connectionPropertiesMap);
+
+    System.clearProperty(HTTP_PROXY_HOST);
```

Comment:
let's run the test code in try and clear the properties in finally or create before method where we always clear the properties (or better assign them to the previous values)